google-cloud-apis/java-apis/maven-apis/src/com/google/cloud/tools/intellij/cloudapis/maven/MavenCloudApiActionDecorator.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.cloudapis.maven;
+
+import com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacet;
+import com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetType;
+import com.google.cloud.tools.intellij.appengine.java.maven.project.MavenProjectService;
+import com.google.cloud.tools.intellij.appengine.java.project.AppEngineProjectService;
+import com.google.cloud.tools.intellij.cloudapis.CloudApiActionDecoratorExtension;
+import com.google.cloud.tools.intellij.cloudapis.GoogleCloudApisMessageBundle;
+import com.intellij.facet.FacetManager;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.JavaVersion;
+
+/**
+ * {@link CloudApiActionDecoratorExtension} extension point that checks for Java projects version if
+ * App Engine is used, and ensures Maven dependencies support is only available for Maven projects.
+ * Otherwise, action is disabled and help text is shown.
+ */
+public class MavenCloudApiActionDecorator implements CloudApiActionDecoratorExtension {
```

Comment:
does the App Engine portion of this need to live here in the Maven module? If we later add Gradle support then the breakdown is a bit odd. But I guess at that point for this to be logically structured we'd need a way for each extension to contribute a check and then aggregate them.

Reply:
Yes, you're right, this extension is rather a transfer of existing functionality in a single piece. It would be better to split it - we don't need Maven check at all (IDEA will do that for us now - Maven functionality will only be available when `maven-api` module is enabled and it depends on Maven presence), and we can move App Engine check into App Engine Java module, and this should allow us to remove this rather unobvious dependency on app engine module. Let me refactor this.